src/components/Queries/DataStatisticsQuery.test.js

```diff
@@ -0,0 +1,271 @@
+import React from 'react'
+import { mount } from 'enzyme'
+import { CustomDataProvider } from '@dhis2/app-runtime'
+import { act } from 'react-dom/test-utils'
+import waitForExpect from 'wait-for-expect'
+import { YEAR } from '../../constants/intervals.js'
+import DataStatisticsQuery from './DataStatisticsQuery.js'
+
+/**
+ * This allows react to update the wrapper after the async app runtime logic has
+ * done its thing and updated state.
+ *
+ * - https://dev.to/dannypule/fix-the-not-wrapped-in-act-warning-simple-solution-3lj1
+ * - https://reactjs.org/docs/testing-recipes.html#act
+ */
+
+const update = wrapper => () =>
+    new Promise(resolve => {
+        setImmediate(() => {
```

Comment:
I'd never heard of this API and support for `setImmediate` seems [very poor](https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate). 

Would it not be possible to use something else for this?

In the olden days we used to just wrap things in `setTimeout` to make sure the DOM was done 😊  :
```js
const update = wrapper => () =>
    new Promise(resolve => {
        setTimeout(() => {
            wrapper.update()
            resolve()
        }, 0)
    })
```
Would that not work?

Reply:
Yeah it's a node thing: https://nodejs.org/api/timers.html#timers_setimmediate_callback_args.

Reply:
You're right in that it's not a standard at all in browsers. Not sure why there's divergence there, but in node I believe it's safe to rely on it.

Reply:
And yeah, I believe `setTimout(cb, 0)` will do something similar as setImmediate (where both will run their callbacks in the next iteration of the event loop). I guess the author of it chose `setImmediate` as it is a little more succinct, since all we want to do is schedule it for the next iteration, and we don't really care about the timeout.

Reply:
Since this is a Jest test which runs in Node it's probably OK here?

Reply:
Yeah I think it's ok to use in node.